### ISSUE TITLE
Simplify Html property for htmlmerger & jsonmerger

### ISSUE THREAD

**Issue by tomalec** [Starcounter/Starcounter#3539]
As discussed on today's meeting with @chrhol, @per-samuelsson, @warpech, @miyconst and me, we agreed to take some steps towards simplifying/cleaning up the flow and code for partials.

We will remove `Html` property from merged partial. Client-side `<starcounter-include>` component will build it from namespaced JSON if there is none in given partial. 
For scopes (individual apps' partials) without `Html` property it will stamp the JSON as text. 
Then many hacks could be removed from server-side code, and the app's `Html` would be the single source of truth - partial HTML components will find a efficient way to render it.

 Launcher's `<starcounter-include>` could wrap HTML-less JSONs into `<juicy-jsoneditor>` for better invulnerability

Sub-issues:
  - Handle nested `Html` property in `<starcounter-include>`
  - Handle nested `Html` property in Launchers `<starcounter-include>`, and use `<juicy-jsoneditor>`
  - Clean the code on server-side

**Comment by chrhol** [comment]
Just adding one point from #3527, the partial URL:s should be encoded to be able to have partial URL:s that contains parameters.

**Comment by chrhol** [comment]
Added issue for serverside tasks and updated the first comment with the link.

**Comment by tomalec** [comment]
https://github.com/Starcounter/starcounter-include/issues/9 implemented, should be available in Nightly `>2.2.1.1995`
@chrhol just to note: URL merged on client-side is not handled by current server-side. It is fully encoded, for example `/starcounter/htmlmerger?Products=%2FProducts%2Fviewmodels%2FProductListPage.html`, and current htmlmerger does not handle `%2F` instead of `/`. 

Moreover, not all pages returns `Html`, like `/Products/viewmodels/EditProductPage.html` do we consider it as Product app bug?

**Comment by warpech** [comment]
> @chrhol just to note: URL merged on client-side is not handled by current server-side. It is fully encoded, for example `/starcounter/htmlmerger?Products=%2FProducts%2Fviewmodels%2FProductListPage.html`, and current htmlmerger does not handle `%2F` instead of `/`. 

@chrhol could you please fix and release a nightly?

> Moreover, not all pages returns `Html`, like `/Products/viewmodels/EditProductPage.html` do we consider it as Product app bug?

Yes. Could you please fix that?

**Comment by chrhol** [comment]
> @chrhol could you please fix and release a nightly?

Yes, although probably not during the day. We have a holiday (a "red" day) here so we're not working. But I have a couple of things I will fix tonight.

**Comment by tomalec** [comment]
> Yes. Could you please fix that?

Sure thing

**Comment by chrhol** [comment]
> But I have a couple of things I will fix tonight.

Looks like I will not be able to do it tonight. I cannot connect to my computer at the office from home, as usual when it's left idle for a while. And I doubt anyone is in the office (to restart it). I will do it first time tomorrow and upload a new nightly after.

**Comment by chrhol** [comment]
I've done these parts today (finally :smile:):
- Decoding the partial url:s on the serverside (and in Launcher). 
- Changed the code in `LauncherHelper.cs` slightly (with the help of @miyconst for avoiding nullreference exceptions) so `IResource.GetHtmlPartialUrl` is no longer used.
- Removed adding the outer `Html` property when serializing json.
- Marked `IResource.GetHtmlPartialUrl` obsolete. Really want to remove it from the interface though, but will leave in there for a while longer.

A new nigthly is currently running, hopefully it will be finished soon and a new Starcounter version will be uploaded. Will post again when it is available.

**Comment by chrhol** [comment]
Nightly with fixes included: http://downloads.starcounter.com/download/Starcounter/DevelopNightlyBuilds/2.2.1.2044
